zerver/lib/actions.py

```diff
@@ -1911,16 +1918,31 @@ def do_send_typing_notification(
 
     send_event(realm, event, user_ids_to_notify)
 
+def check_send_stream_typing_notification(sender: UserProfile, operator: str,
+                                          stream_id: int, topic: str) -> None:
+    realm = sender.realm
+    if operator not in ('start', 'stop'):
+        raise JsonableError(_('Invalid \'op\' value (should be start or stop)'))
+
+    try:
+        get_stream_by_id_in_realm(stream_id, realm)
```

Comment:
Let's use the `access_stream_...` methods, called from the view function -- that's our standard approach to security.

I think it's also wise to do a preparatory refactoring commit to move the `start|stop` validation into the view function -- I think you can do it in `REQ` with `check_string_in`.

Reply:
Changed to do validation using `access_stream_by_id`.

Yes, It can be done using `REQ` with `check_string_in`. Added a preparatory commit. One thing to notice here is doing it in `REQ` would raise error just saying `Invalid op`but we'll miss the verbose error `Invalid 'op' value (should be start or stop` we were returning.